source/s3_auto_ranged_put.c

```diff
@@ -548,8 +548,16 @@ static bool s_s3_auto_ranged_put_update(
 
             if (should_create_next_part_request) {
 
-                struct aws_s3_buffer_pool_ticket *ticket =
-                    aws_s3_buffer_pool_reserve(meta_request->client->buffer_pool, meta_request->part_size);
+                struct aws_s3_buffer_pool_ticket *ticket = NULL;
+                if (meta_request->synced_data.async_write.ready_to_send) {
+                    /* Async-write already has a ticket, take ownership */
+                    AWS_ASSERT(meta_request->synced_data.async_write.buffered_data_ticket);
```

Comment:
should we make it a fatal? it would fail hard if we dont have a ticket, right?

Reply:
sure. it's a cheap check